src/devices/Bmp280/Bmp280.cs

```diff
@@ -0,0 +1,562 @@
+            //Convert the pressure to Hectopascals(hPa)
+            pressure /= 100;
+
+            //Calculate and return the altitude using the international barometric formula
+            return 44330.0 * (1.0 - Math.Pow((pressure / seaLevelPressure), 0.1903));
+        }
+
+        /// <summary>
+        ///  Returns the temperature in DegC. Resolution is 0.01 DegC. Output value of “5123” equals 51.23 DegC.
+        /// </summary>
+        /// <param name="adcT"></param>
+        /// <returns>
+        ///  Degrees celsius
+        /// </returns>
+        private double BMP280_compensate_T_double(int adcT)
+        {
+            //The temperature is calculated using the compensation formula in the BMP280 datasheet
+            var var1 = ((adcT / 16384.0) - (_calibrationData.DigT1 / 1024.0)) * _calibrationData.DigT2;
+            var var2 = ((adcT / 131072.0) - (_calibrationData.DigT1 / 8192.0)) * _calibrationData.DigT3;
+
+            _temperatureFine = (int)(var1 + var2);
+
+            var T = (var1 + var2) / 5120.0;
+            return T;
+        }
+
+        /// <summary>
+        ///  Returns the pressure in Pa, in Q24.8 format (24 integer bits and 8 fractional bits).
+        ///  Output value of “24674867” represents 24674867/256 = 96386.2 Pa = 963.862 hPa
+        /// </summary>
+        /// <param name="adcP"></param>
+        /// <returns>
+        ///  Pressure in hPa
+        /// </returns>
+        private long BMP280_compensate_P_Int64(int adcP)
+        {
+            //The pressure is calculated using the compensation formula in the BMP280 datasheet
+            long var1 = _temperatureFine - 128000;
+            var var2 = var1 * var1 * (long)_calibrationData.DigP6;
+            var2 = var2 + ((var1 * (long)_calibrationData.DigP5) << 17);
+            var2 = var2 + ((long)_calibrationData.DigP4 << 35);
+            var1 = ((var1 * var1 * (long)_calibrationData.DigP3) >> 8) + ((var1 * (long)_calibrationData.DigP2) << 12);
+            var1 = (((((long)1 << 47) + var1)) * (long)_calibrationData.DigP1) >> 33;
+            if (var1 == 0)
+            {
+                return 0; //Avoid exception caused by division by zero
+            }
+            //Perform calibration operations as per datasheet: http://www.adafruit.com/datasheets/BST-BMP280-DS001-11.pdf
+            long p = 1048576 - adcP;
+            p = (((p << 31) - var2) * 3125) / var1;
+            var1 = ((long)_calibrationData.DigP9 * (p >> 13) * (p >> 13)) >> 25;
+            var2 = ((long)_calibrationData.DigP8 * p) >> 19;
+            p = ((p + var1 + var2) >> 8) + ((long)_calibrationData.DigP7 << 4);
+            return p;
+        }
+
+        /// <summary>
+        ///  Reads an 8 bit value from a register
+        /// </summary>
+        /// <param name="register">
+        ///  Register to read from
+        /// </param>
+        /// <returns>
+        ///  Value from register
+        /// </returns>
+        public byte Read8(byte register)
```

Comment:
I would make these methods private, as they are only really used inside your binding, and might just confuse consumers when they are looking for the higher level methods like ReadTemperature or ReadPreassure. Also, I know that this is how they were named in Adafruit's repo, but can we rename them to either `ReadByte` `ReadShort` or at least something like `Read8BitsFromRegister` `Read16BitsFromRegister`? That will just make it feel more like .NET and more close according to the framework.

Reply:
Fixed